### PR TITLE
Remove --pre on generated sources

### PR DESCRIPTION
**Pull request by jensens**
also address #38 while at it

### CODE REVIEW

File: src/mxdev/processing.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def write_dev_sources(fio, packages: typing.Dict[str, typing.Dict[str, typing.An
             continue
         extras = f"[{package['extras']}]" if package["extras"] else ""
         subdir = f"/{package['subdirectory']}" if package["subdirectory"] else ""
-        install_options = " --pre"
+        install_options = ""
```
Comment (rnixx) [inline src/mxdev/processing.py:208]: empty install options? imo they can be removed entirely

Comment (jensens) [reply src/mxdev/processing.py:208]: well, yes, why not

File: CHANGES.md [hunk 2]
```diff
@@ -9,6 +9,9 @@
   This change enables the use of the generated file with the alternative pip replacement `uv`.
   [jensens]
 
+- Breaking: Drop support for Python 3.7 (it is end of life).
```
Comment (rnixx) [inline CHANGES.md:12]: - Remove python 3.7 from CI. It still works but may break in future versions

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -11,14 +11,13 @@ jobs:
...
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
       - uses: pre-commit/action@v3.0.0
   build:
     strategy:
       matrix:
         python-version:
-          - "3.7"
```
Comment (jensens) [inline .github/workflows/tests.yaml:21]: did it

**Pull request merged**